compiler/resolution/lowerIterators.cpp

```diff
@@ -1610,6 +1610,21 @@ inlineSingleYieldIterator(ForLoop* forLoop) {
   forLoop->replace(body);
 }
 
+
+// When expanding a single-loop iterator with a single yield, the generated
```

Comment:
This comment is too high up. expandForLoop does regular iterator inlining if it can and only falls back to using this form in the else block